components/movementsensor/imuvectornav/imu.go

```diff
@@ -158,7 +186,10 @@ func newVectorNav(
 		fwver[2],
 		fwver[3],
 	)
+	return nil
+}
 
+func (vn *vectornav) imuConfig(ctx context.Context) error {
```

Comment:
```suggestion
func (vn *vectornav) configureVectorNav(ctx context.Context) error {
```
but I'd rather this be in the constructor.

Reply:
I am slightly confused by this comment. Above you say that you prefer a "flatter" version of constructor, but here you give me a naming suggestion. Do you want me to make the constructor flatter while also keeping this helper function?

Reply:
Two things are true:
- the name of this function does not currently nicely describe what it is doing.
- I prefer flat constructors, but I know susmita does not, so that's up to you. For me the happy path is not having structs populated by helper functions, or the stoppable workers started from within a helper function (rather, have the stoppable workers consume a helper function after everything in the imu struct has been set. I don't find the extra layers helpful from a debugging perspective.

Reply:
Just to be clear, we can add helpers, but I do prefer that helpers are designed to be isolated and testable and have a reason.

Reply:
You could also, in general, approach tickets which need a refactor like so:
- address the bug first, add a new pr refactoring
- prefactor the code to make it more readable (limit such tickets to 1-2 days), address the bug in a follow up pr.

Reply:
> Just to be clear, we can add helpers, but I do prefer that helpers are designed to be isolated and testable and have a reason.

Them being testable is a good point.
I do not know how any of the helper functions could be tested since we would need to interact with the imu itself.
For the reason of the helpers not being testable I will revert back the constructor so that it is completely flat.